jsapp/js/account/organization/OrganizationSettingsRoute.tsx

```diff
@@ -1,7 +1,156 @@
-import React from 'react';
+// Libraries
+import {useState, useEffect} from 'react';
 
+// Partial components
+import OrganizationSettingsField from './OrganizationSettingsField';
+import LoadingSpinner from 'jsapp/js/components/common/loadingSpinner';
+import InlineMessage from 'jsapp/js/components/common/inlineMessage';
+import Button from 'jsapp/js/components/common/button';
+
+// Stores, hooks and utilities
+import useWhenStripeIsEnabled from 'js/hooks/useWhenStripeIsEnabled.hook';
+import {OrganizationUserRole, useOrganizationQuery} from 'js/account/organization/organizationQuery';
+import subscriptionStore from 'js/account/subscriptionStore';
+import envStore from 'js/envStore';
+import {getSimpleMMOLabel} from './organization.utils';
+
+// Constants and types
+import {ORGANIZATION_TYPE_SELECT_OPTIONS} from 'js/account/accountFieldsEditor.component';
+
+// Styles
+import styles from 'js/account/organization/organizationSettingsRoute.module.scss';
+
+interface State {
+  name: string;
+  website?: string;
+  type?: string;
+}
+
+/**
+ * Renders few fields with organization related settings, like name or website
+ * (with some logic in regards to their visibility). If user has necessary role,
+ * they can edit available fields.
+ */
 export default function OrganizationSettingsRoute() {
+  const orgQuery = useOrganizationQuery();
+  const [subscriptions] = useState(() => subscriptionStore);
+  const [state, setState] = useState<State>({name: ''});
```

Comment:
I'm not really a fan of this pattern of having an object in state. 
Usually when that's the case it's common to use `useReducer`, but in this case, just for a few fields I'd go with multiple states.